### ISSUE TITLE
[BUG] NumbaWarning: The TBB threading layer requires TBB version 2019.5 or later i.e., TBB_INTERFACE_VERSION >= 11005.

### ISSUE THREAD

**Issue by lc822** [aertslab/pySCENIC#182]
**Describe the bug**
NumbaWarning: The TBB threading layer requires TBB version 2019.5 or later i.e., TBB_INTERFACE_VERSION >= 11005. Found TBB_INTERFACE_VERSION = 6103. The TBB threading layer is disabled.

I am getting this warning when I run arboreto_with_multiprocessing.py in a Python virtual environment with `tbb==2020.0.133`

**Steps to reproduce the behavior**
1. Command run when the error occurred:

```python
python arboreto_with_multiprocessing.py --output adjacencies.tsv --num_workers 24 raw_counts.csv lambert2018.txt
```

2. Error encountered:

```pytb
NumbaWarning: The TBB threading layer requires TBB version 2019.5 or later i.e., TBB_INTERFACE_VERSION >= 11005. Found TBB_INTERFACE_VERSION = 6103. The TBB threading layer is disabled.
```

**Expected behavior**
Use of TBB threading

**Please complete the following information:**
- pySCENIC version: 0.10.2
- Installation method: Pip
- Run environment: CLI
- OS: Red Hat
- Package versions: [obtain using `pip freeze`, `conda list`, or skip this if using Docker/Singularity]:

```
anndata==0.7.3
apipkg==1.5
arboreto==0.1.5
attrs==19.3.0
backcall==0.2.0
bcrypt==3.1.7
bleach==3.1.5
boltons==20.1.0
certifi==2020.4.5.2
cffi==1.14.0
-e git+https://github.com/cgat-developers/cgat-core.git@97cb7f49f1cf337f8d0d685c5b9d2548a49ce6c7#egg=cgatcore
chardet==3.0.4
click==7.1.2
cloudpickle==1.4.1
cryptography==2.9.2
cycler==0.10.0
cytoolz==0.10.1
dask==1.0.0
decorator==4.4.2
defusedxml==0.6.0
dill==0.3.2
distributed==1.28.1
drmaa==0.7.9
entrypoints==0.3
execnet==1.7.1
frozendict==1.2
get-version==2.1
gevent==20.6.1
greenlet==0.4.16
h5py==2.10.0
HeapDict==1.0.1
idna==2.9
importlib-metadata==1.6.1
interlap==0.2.6
ipykernel==5.3.0
ipython==7.15.0
ipython-genutils==0.2.0
ipywidgets==7.5.1
jedi==0.17.0
Jinja2==2.11.2
joblib==0.15.1
json5==0.9.5
jsonschema==3.2.0
jupyter==1.0.0
jupyter-client==6.1.3
jupyter-console==6.1.0
jupyter-core==4.6.3
jupyterlab==2.1.4
jupyterlab-server==1.1.5
kiwisolver==1.2.0
legacy-api-wrap==1.2
llvmlite==0.33.0
loompy==3.0.6
MarkupSafe==1.1.1
matplotlib==3.2.2
mistune==0.8.4
more-itertools==8.3.0
msgpack==1.0.0
multiprocessing-on-dill==3.5.0a4
natsort==7.0.1
nbconvert==5.6.1
nbformat==5.0.7
networkx==2.4
notebook==6.0.3
numba==0.50.0
numexpr==2.7.1
numpy==1.18.5
numpy-groupies==0+unknown
packaging==20.4
pandas==0.25.3
pandocfilters==1.4.2
paramiko==2.7.1
parso==0.7.0
patsy==0.5.1
pep8==1.7.1
pexpect==4.8.0
pickleshare==0.7.5
pluggy==0.13.1
prometheus-client==0.8.0
prompt-toolkit==3.0.5
psutil==5.7.0
ptyprocess==0.6.0
py==1.8.1
pyarrow==0.16.0
pycparser==2.20
Pygments==2.6.1
PyNaCl==1.4.0
pyparsing==2.4.7
pyrsistent==0.16.0
pyscenic==0.10.2
pytest==5.4.3
pytest-cache==1.0
pytest-pep8==1.0.6
python-dateutil==2.8.1
pytz==2020.1
PyYAML==5.3.1
pyzmq==19.0.1
qtconsole==4.7.4
QtPy==1.9.0
requests==2.23.0
ruffus==2.8.4
scanpy==1.5.1
scikit-learn==0.23.1
scipy==1.4.1
seaborn==0.10.1
Send2Trash==1.5.0
setuptools-scm==4.1.2
six==1.15.0
sortedcontainers==2.2.2
SQLAlchemy==1.3.17
statsmodels==0.11.1
tables==3.6.1
tbb==2020.0.133
tblib==1.6.0
terminado==0.8.3
testpath==0.4.4
threadpoolctl==2.1.0
toolz==0.10.0
tornado==6.0.4
tqdm==4.46.1
traitlets==4.3.3
umap-learn==0.4.4
urllib3==1.25.9
wcwidth==0.2.4
webencodings==0.5.1
widgetsnbextension==3.5.1
zict==2.0.0
zipp==3.1.0
zope.event==4.4
zope.interface==5.1.0
```

**Comment by nneubert** [comment]
I just barely ran into this same issue while trying to implement umap.

**Comment by cflerin** [comment]
Interesting. But I can't reproduce this error with the same versions of your numpy and tbb packages. I would guess that there is something wrong with your numba installation. Could you check the output of:
```
numba -s
```
specifically the section on threading:
```
__Threading Layer Information__
TBB Threading layer available                 : True
OpenMP Threading layer available              : False
+--> Disabled due to                          : Unknown import problem.
Workqueue Threading layer available           : True
```

**Comment by lc822** [comment]
Thank you, this is what I get:

```
__Threading Layer Information__
TBB Threading Layer Available                 : True
+-->TBB imported successfully.
OpenMP Threading Layer Available              : False
+--> Disabled due to Unknown import problem.
Workqueue Threading Layer Available           : True
+-->Workqueue imported successfully.
```

**Comment by cflerin** [comment]
Closing this since it seems to be a numba install issue

**Comment by dmiyagi** [comment]
Hi @cflerin , sorry to open this back up but I got the same error but the threading info is different demonstrating that my TBB is disabled. It's on an HPC that I don't have admin permissions to so hopefully its resolvable without that: 

> __Threading Layer Information__
TBB Threading Layer Available                 : False
+--> Disabled due to undefined symbol: _ZN3tbb10interface78internal15task_arena_base19internal_initializeEv
OpenMP Threading Layer Available              : True
+-->Vendor: GNU
Workqueue Threading Layer Available           : True
+-->Workqueue imported successfully.


appreciate any help!

**Comment by cflerin** [comment]
@dmiyagi ,

Sorry, I haven't ever seen this error... I would still guess that it's an install problem, especially since you don't have admin access (although this has worked fine for me in a conda environment in the same situation). Could you potentially try the Docker image? Reinstall in a clean conda environment? Or try one of the additional packages suggested [here](https://numba.pydata.org/numba-doc/dev/user/threading-layer.html)?

**Comment by dmiyagi** [comment]
Hi thanks for your response! i'll give your recommendations a try and report back in case anyone else needs the info down the road.

**Comment by Jintram** [comment]
I ran into the same error, I think I have a workaround.

Leaving this comment for other people with the same issue.

When running 
pyscenic grn {exprmatrix.loom} {TF_file} -o adj.csv --num_workers 20
I got the error message:
~/mwehrens/bin/miniconda3/envs/pyscenic/lib/python3.7/site-packages/numba/np/ufunc/parallel.py:365: NumbaWarning: The TBB threading layer requires TBB version 2019.5 or later i.e., TBB_INTERFACE_VERSION >= 11005. Found TBB_INTERFACE_VERSION = 6103. The TBB threading layer is disabled.
  warnings.warn(problem)
  
I'm not an expert in any of these matter, but it seems to be an issue of numba and its dependency TBB.

There's an issue reported at numba here:
https://github.com/numba/numba/issues/6350
And using the code shown there, I can also reproduce the warning on my system without any SCENIC code, so it's indeed not a SCENIC issue.

Luckily also one of the suggested solutions worked. So I executed the following command:
NUMBA_THREADING_LAYER='omp' pyscenic grn {exprmatrix.loom} {TF_file} -o adj.csv --num_workers 20

Which then ran apparently successfully without showing the warning.
